scripts/auto-rebase/rebase.sh

```diff
@@ -43,6 +42,159 @@ title() {
+# operator bundle is the authoritative source for manifest and image digests.
+download_lvms_operator_bundle(){
+    title 'downloading LVMS operator bundles'
+    declare -A images=(["amd64"]="${1}" ["arm64"]="${2:-}")
+    local LVMS_STAGING="${STAGING_DIR}/lvms"
+    for arch in "${!images[@]}"; do
+        # TODO special case arm64 builds of topolvm since they are not available in 4.12
+        if [[ "$arch" == "arm64" && "${images[${arch}]}" == "" ]]; then
+            continue
+        fi
+        img="${images[$arch]}"
+        mkdir -p "$LVMS_STAGING/$arch"
+        pushd "$LVMS_STAGING/$arch" || return 1
+        oc image extract \
+            --registry-config "$PULL_SECRET_FILE" \
+            --path /manifests/:. "$img" \
+            --filter-by-os "$arch" \
+            ||  {
+                    popd
+                    return 1
+                }
+        popd || return 1
+      done
+}
+
+parse_images() {
+    local src=$1
+    yq '.spec.relatedImages[]? | [.name, .image] | @csv' $src > "$(dirname "$src")/images"
+}
+
+write_lvms_images_for_arch(){
+
+    declare -a include_images=(
+        "topolvm-csi"
+        "topolvm-csi-provisioner"
+        "topolvm-csi-resizer"
+        "topolvm-csi-registrar"
+        "topolvm-csi-livenessprobe"
+    )
+
+    local arch="$1"
+    [ -d "$arch" ] || {
+        echo "dir $arch not found"
+        return 1
+    }
+    local csv_manifest="${arch}/lvms-operator.clusterserviceversion.yaml"
+
+    parse_images "$csv_manifest"
+    image_file="./${arch}/images"
+    if ! [ -e $image_file ]; then
+        echo "image file \"$image_file\" not found"
+        return 1
+    fi
+    while read -ers LINE; do
+        name=${LINE%,*}
+        img=${LINE#*,}
+        found=false
+        for included in "${include_images[@]}"; do
+            if [[ "$name" == $included ]]; then
+                found=true
+                break;
+            fi
+        done
+        $found || continue # name not ’found in $REPOROOT/assets/release/release-$arch.json, skip
+        yq -iP -o=json e '.images["'"$name"'"] = "'"$img"'"' "${REPOROOT}/assets/release/release-${GOARCH_TO_UNAME_MAP[${arch}]}.json"
+    done < "$image_file"
+}
+
+update_lvms_images(){
+    local workdir="$STAGING_DIR/lvms"
+    [ -d "$workdir" ] || {
+        >&2 echo 'lvms staging dir not found, aborting image update'
+        return 1
+    }
+    pushd "$workdir"
+    local archs
+    set -x
+    archs="$(ls -d */ | sed 's,/$,,')"
+    for a in $archs; do
+        write_lvms_images_for_arch "$a"
+    done
+    popd
+}
+
+set_openshift_storage_ns(){
+        local manifest="$1"
+        local tmp="$(oc create --dry-run=client --namespace=openshift-storage -o yaml -f "${manifest}")"
+        # assume that if the returned string is nil, it's because the resource is not namespaces.  Failures
+        # like file-not-found will still result in the script exiting with a non-zero code.
```

Comment:
Can you explain what we're trying to do here? We are taking a manifest and overwriting it eventually?

Reply:
The manifests in the operator bundle don't specify namespace, and MicroShift doesn't inherently assume a namespace for LVMS workloads.  This command feeds the downloaded manifests through `oc` to inject the namespace and outputs new yamls.  Using `oc` is nice b/c it won't specify a namespace on a non-namespaced object.

Reply:
Is `oc` available in the rebase script when it runs in CI?

Reply:
Yes, it's an existing prereq for the script.  This PR does not add new dependencies.